cel/program.go

```diff
@@ -125,20 +126,29 @@ func newCtxResult(val ref.Val, det *EvalDetails, err error) *ctxResult {
 	}
 }
 
+func (ed *EvalDetails) ActualCost() *uint64 {
+	if ed.costTracker == nil {
+		return nil
+	}
+	cost := ed.costTracker.ActualCost()
+	return &cost
+}
+
 // prog is the internal implementation of the Program interface.
 type prog struct {
 	*Env
-	evalOpts      EvalOption
-	decorators    []interpreter.InterpretableDecorator
-	defaultVars   interpreter.Activation
-	dispatcher    interpreter.Dispatcher
-	interpreter   interpreter.Interpreter
-	interpretable interpreter.Interpretable
-	attrFactory   interpreter.AttributeFactory
+	evalOpts          EvalOption
+	decorators        []interpreter.InterpretableDecorator
+	defaultVars       interpreter.Activation
+	dispatcher        interpreter.Dispatcher
+	interpreter       interpreter.Interpreter
+	interpretable     interpreter.Interpretable
+	attrFactory       interpreter.AttributeFactory
+	callCostEstimator interpreter.FunctionActualCostEstimator
```

Comment:
nit: type name is long, maybe just "ActualCostEstimator" ?  I'm okay with keeping the "Estimator" suffix because for functions we're still making what is clearly an estimate.